src/main/java/org/shanerx/tradeshop/utils/Updater.java

```diff
@@ -0,0 +1,190 @@
+					log.log(Level.WARNING, "[Updater] Please update asap from: https://goo.gl/Qv2iVZ");
+					log.log(Level.WARNING, "[Updater] +------------------------------------------------+");
+					in.close();
+					return RelationalStatus.BEHIND;
+				} else if (rs == RelationalStatus.AHEAD) {
+					log.log(Level.WARNING, "[Updater] +-----------------------------------------------------+");
+					log.log(Level.WARNING, "[Updater] You are running a developmental version of the plugin!");
+					log.log(Level.WARNING, "[Updater] Most recent stable version: " + inputLine);
+					log.log(Level.WARNING, "[Updater] Please notice that the build may contain critical bugs!");
+					log.log(Level.WARNING, "[Updater] +-----------------------------------------------------+");
+					in.close();
+					return RelationalStatus.AHEAD;
+				} else {
+					log.log(Level.WARNING, "[Updater] You are running the latest version of the plugin!");
+					in.close();
+					return RelationalStatus.UP_TO_DATE;
+				}
+			}
+		} catch (Exception e) {
+			e.printStackTrace();
+			log.log(Level.WARNING, "[Updater] +----------------------------------------------------+");
+			log.log(Level.WARNING, "[Updater] Could not establish a connection to check for updates!");
+			log.log(Level.WARNING, "[Updater] +----------------------------------------------------+");
+		}
+		return RelationalStatus.UNKNOWN;
+	}
+
+	public RelationalStatus compareVersions(final String major, final String minor, final String patch) {
+		try {
+			return compareVersions(Short.parseShort(major), Short.parseShort(minor), Short.parseShort(patch));
+		} catch (NumberFormatException e) {
+			throw new IllegalArgumentException("All arguments must be numbers!");
+		}
+	}
+
+	public RelationalStatus compareVersions(final short major, final short minor, final short patch) {
+		try {
+			short selfMajor = Short.parseShort(getVersionComponent(SemVer.MAJOR));
+			short selfMinor = Short.parseShort(getVersionComponent(SemVer.MINOR));
+			short selfPatch = Short.parseShort(getVersionComponent(SemVer.PATCH));
+
+			if (selfMajor == major && selfMinor == minor && selfPatch == patch) {
+				return RelationalStatus.UP_TO_DATE;
+			} else if (selfMajor > major) {
+				return RelationalStatus.AHEAD;
+			} else if (selfMajor < major) {
+				return RelationalStatus.BEHIND;
+			} else if (selfMinor > minor) {
+				return RelationalStatus.AHEAD;
+			} else if (selfMinor < minor) {
+				return RelationalStatus.BEHIND;
+			} else if (selfPatch > patch) {
+				return RelationalStatus.AHEAD;
+			} else {
+				return RelationalStatus.BEHIND;
+			}
+
+		} catch (NumberFormatException e) {
+			throw new IllegalArgumentException("All arguments must be numbers!");
+		}
+	}
+
+	public enum BuildType {
+
+		DEV,
+
+		BETA,
+
+		STABLE,
+
+		FINAL
+	}
+
+	public enum SemVer {
+
+		MAJOR,
+
+		MINOR,
+
+		PATCH,
+
+		BUILD_TYPE
+	}
+
+	public enum RelationalStatus {
+
+		AHEAD,
+
+		UP_TO_DATE,
+
+		BEHIND,
+
+		UNKNOWN
```

Comment:
those empty lines are painful to look at, but not the point.
Is there anything new here? this file was moved, and the previous one had fewer lines.

Reply:
Moved i think is all, also you put those in there :P
Fixed now tho.

Reply:
I am aware of the inconsistency in my own code in the past. Still would like to figure out what was added to the file.